### PR TITLE
Prevent to add PageInfo definition if not needed

### PR DESCRIPTION
**Pull request by ooga**
## Checklist

- [x] Pull requests follows the [contribution guide](https://github.com/graphql-java-kickstart/graphql-java-tools/wiki/Contribution-guide)
- [x] New or modified functionality is covered by tests

## Description

Prevent `RelayConnectionFactory` to do extra work and add new definition when not needed.

### CODE REVIEW

**Comment by oryan-block** [comment]
Can you provide an example schema where this is a problem?
I'm not sure I understand what we're trying  to solve here.

**Comment by vojtapol** [comment]
I believe the problem is that we are always adding some baked-in types to facilitate the relay `@connection` directive. If the users want to have their own type called `PageInfo` for whatever reason they can't because it clashes.

File: src/test/kotlin/graphql/kickstart/tools/relay/RelayConnectionFactoryTest.kt [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    @Test
+    fun `should not add new definition when no @connection directive`() {
+        // setup
+        val factory = RelayConnectionFactory()
+        val existing = mutableListOf<Definition<*>>()
+
+        val newDefinitions = factory.create(existing)
+
+        // expect
+        Assert.assertEquals(newDefinitions.size, 0)
```
Comment (vojtapol) [inline src/test/kotlin/graphql/kickstart/tools/relay/RelayConnectionFactoryTest.kt:18]: Please static-import `assertEquals`.

File: src/main/kotlin/graphql/kickstart/tools/relay/RelayConnectionFactory.kt [hunk 2]
```diff
@@ -6,12 +6,17 @@ import graphql.language.*
 class RelayConnectionFactory : TypeDefinitionFactory {
 
     override fun create(existing: MutableList<Definition<*>>): List<Definition<*>> {
+        val connectionDirectives = findConnectionDirectives(existing)
+        if (connectionDirectives.isEmpty()) {
```
Comment (vojtapol) [inline src/main/kotlin/graphql/kickstart/tools/relay/RelayConnectionFactory.kt:10]:
```suggestion
        if (connectionDirectives.isEmpty()) {
            // do not add Relay definitions unless needed
```

**Review by vojtapol — CHANGES_REQUESTED** [review]
Thanks for the contribution!

**Review by oryan-block — APPROVED** [review]

**Review by vojtapol — APPROVED** [review]

**Pull request merged**